src/Microsoft.Tye.Core/ProjectReader.cs

```diff
@@ -30,94 +39,136 @@ public static async Task ReadProjectDetailsAsync(OutputContext output, FileInfo
+            project.Version = version;
+            output.WriteDebugLine($"Found application version: {version}");
+
+            var targetFramework = projectInstance.GetPropertyValue("TargetFramework");
+            project.TargetFramework = targetFramework;
+            output.WriteDebugLine($"Found target framework: {targetFramework}");
+
+            var sharedFrameworks = projectInstance.GetItems("FrameworkReference").Select(i => i.EvaluatedInclude).ToList();
+            project.Frameworks.AddRange(sharedFrameworks.Select(s => new Framework(s)));
+            output.WriteDebugLine($"Found shared frameworks: {string.Join(", ", sharedFrameworks)}");
+
+            output.WriteDebugLine($"Evaluation Took: {sw.Elapsed.TotalMilliseconds}ms");
+
+            // The Microsoft.Build.Locator doesn't handle the loading of other assemblies
+            // that are shipped with MSBuild (ex NuGet).
+            //
+            // This means that the set of assemblies that need special handling depends on the targets
+            // that we run :(
+            //
+            // This is workaround for this limitation based on the targets we need to run
+            // to resolve references and versions.
+            //
+            // See: https://github.com/microsoft/MSBuildLocator/issues/86
+            Assembly? ResolveAssembly(AssemblyLoadContext context, AssemblyName assemblyName)
+            {
+                if (assemblyName.Name is object && assemblyName.Name.StartsWith("NuGet."))
                 {
-                    var line = lines[i];
-                    if (line.StartsWith("version="))
+                    var msbuildDirectory = Environment.GetEnvironmentVariable("MSBuildExtensionsPath")!;
+                    var assemblyFilePath = Path.Combine(msbuildDirectory, assemblyName.Name + ".dll");
+                    if (File.Exists(assemblyFilePath))
                     {
-                        project.Version = line.Substring("version=".Length).Trim();
-                        output.WriteDebugLine($"Found application version: {line}");
-                        continue;
-                    }
-
-                    if (line.StartsWith("tfm"))
-                    {
-                        project.TargetFramework = line.Substring("tfm=".Length).Trim();
-                        output.WriteDebugLine($"Found target framework: {line}");
-                        continue;
-                    }
-
-                    if (line.StartsWith("frameworks="))
-                    {
-                        var right = line.Substring("frameworks=".Length).Trim();
-                        project.Frameworks.AddRange(right.Split(",").Select(s => new Framework(s)));
-                        output.WriteDebugLine($"Found shared frameworks: {line}");
-                        continue;
+                        return Assembly.LoadFrom(assemblyFilePath);
```

Comment:
Any reason to avoid using context.LoadFromAssemblyPath? It won't load deps but you'll likely be called for those anyways?

Reply:
I'll try and it and see. That's a bit more primitive so I don't mind

Reply:
ya it works 👍